File: python/resistor-color-duo/resistor_color_duo.py
ALL_COLORS = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
]


def value(colors: list[str]) -> int:
    number_value = ""
    for color in colors[:2]:
        if color not in ALL_COLORS:
            raise ValueError("unexpected error of color")

        pos = ALL_COLORS.index(color)

        number_value += str(pos)

    return int(number_value)

File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.Arrays;
import java.util.List;

class ResistorColorDuo {
  private final String[] allColors = {
    "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"
  };

  String colorValue(String color) {
    return Integer.toString(Arrays.asList(allColors).indexOf(color));
  }

  int value(String[] colors) {
    String result = "";

    List<String> listOfTwo = Arrays.asList(colors).subList(0, 2);

    for (int i = 0; i < listOfTwo.size(); i++) {
      result += colorValue(listOfTwo.get(i));
    }

    return Integer.parseInt(result);
  }
}
